test/Shared/UserManagerTestBase.cs

```diff
@@ -1208,7 +1205,7 @@ public Task<IdentityResult> ValidateAsync(UserManager<TUser> manager, TUser user
                 Assert.Equal(roles.Count, rs.Count);
                 foreach (var r in roles)
                 {
-                    Assert.True(rs.Any(role => role == r.Name));
+                    Assert.True(rs.Any(role => role == roleManager.GetRoleNameAsync(r).Result));
```

Comment:
See other comments about `.Result`. Let's figure out if we can something better.